### PR TITLE
Alonzo audit

### PR DESCRIPTION
**Pull request by WhatisRT**
Lots of small changes/additions pointed out by different audits

### CODE REVIEW

**Review by JaredCorduan — APPROVED** [review]
looks good to me!

File: alonzo/formal-spec/utxo.tex [hunk 1]
```diff
@@ -8,23 +8,26 @@ \section{UTxO}
...
+    & \fun{totExUnits} \in \Tx \to \ExUnits \\
+    & \fun{totExUnits}~\var{tx} = \sum_{r \in \fun{txrdmrs}~tx} \fun{exunits}~r
     \nextdef
     & \fun{feesOK} \in \PParams \to \Tx \to \UTxO \to \Bool  \\
     & \fun{feesOK}~\var{pp}~tx~utxo~= \\
     &~~      \minfee{pp}~{tx} \leq \txfee{txb} \wedge (\fun{txrdmrs}~tx \neq \Nothing \Rightarrow \\
-    &~~~~~~((\forall (a, \wcard, \_) \in \fun{range}~(\fun{collateral}~{txb} \restrictdom \var{utxo}), \fun{paymentHK}~a \in \AddrVKey) \\
+    &~~~~~~((\forall (a, \wcard, \_) \in \fun{range}~(\fun{collateral}~{txb} \restrictdom \var{utxo}), a \in \AddrVKey) \\
     &~~~~~~\wedge~ \fun{adaOnly}~\var{balance} \\
     &~~~~~~\wedge~ \var{balance}~*~100 \geq \txfee~{txb} * (\fun{collateralPercent}~pp))) \\
```
Comment (polinavino) [inline alonzo/formal-spec/utxo.tex:23]: i think this has a couple extra brackets? I'm having a hard time tracking them here hehe

File: alonzo/formal-spec/utxo.tex [hunk 2]
```diff
@@ -8,23 +8,26 @@ \section{UTxO}
...
-    &~~~~~~\wedge~ \fun{collateral}~{tx}~ \neq~\{\} \\
+    &~~~~~~\wedge~ \fun{collateral}~{tx}~ \neq~\{\}) \\
```
Comment (polinavino) [inline alonzo/formal-spec/utxo.tex:24]: i think this closes the line 21 first bracket?

File: alonzo/formal-spec/utxo.tex [hunk 3]
```diff
@@ -659,7 +663,7 @@ \subsection{The UTXOS transition system}
       \var{txb}\leteq\txbody{tx} &
       \fun{ininterval}~\var{slot}~(\fun{txvldt}~{tx}) &
       \hldiff{\var{(\wcard, i_f)}\leteq\fun{txvldt}~{tx}} \\~\\
-      \hldiff{\fun{txrdmrs}~\var{tx}\neq \Nothing ~\Rightarrow~ \fun{epochInfoSlotToUTCTime}~\mathsf{epochInfo}~\mathsf{systemTime}~i_f \neq \Nothing} \\
+      \hldiff{\Nothing \notin \{\fun{txrdmrs}~\var{tx}, i_f\} ~\Rightarrow~ \fun{epochInfoSlotToUTCTime}~\mathsf{epochInfo}~\mathsf{systemTime}~i_f \neq \Nothing} \\
```
Comment (polinavino) [inline alonzo/formal-spec/utxo.tex:666]: \notin?

**Comment by polinavino** [comment]
two tiny things (which maybe Im just confused about, looks really good otherwise)

File: alonzo/formal-spec/utxo.tex (hunk 3 above)
Comment (WhatisRT) [reply alonzo/formal-spec/utxo.tex:666]: It's a compact way of saying that both of those aren't Nothing. So this modifies the line to only check the conversion if `i_f` isn't nothing as well

File: alonzo/formal-spec/utxo.tex (hunk 2 above)
Comment (WhatisRT) [reply alonzo/formal-spec/utxo.tex:24]: Yeah, I think the parentheses are fine

Comment (polinavino) [reply alonzo/formal-spec/utxo.tex:24]: ok

File: alonzo/formal-spec/utxo.tex (hunk 3 above)
Comment (polinavino) [reply alonzo/formal-spec/utxo.tex:666]: right, didnt notice it was \notin ... a set

**Pull request merged**